sql/catalyst/src/test/scala/org/apache/spark/sql/RowJsonSuite.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import java.sql.{Date, Timestamp}
+import java.time.{Instant, LocalDate}
+
+import org.json4s.JsonAST.{JArray, JBool, JDecimal, JDouble, JLong, JNull, JObject, JString, JValue}
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.encoders.{ExamplePoint, ExamplePointUDT}
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types._
+
+/**
+ * Test suite for [[Row]] JSON serialization.
+ */
+class RowJsonSuite extends SparkFunSuite {
+  private val schema = new StructType()
+    .add("c1", "string")
+    .add("c2", IntegerType)
+
+  private val conf: SQLConf = {
+    val conf = new SQLConf
+    conf.setConf(SQLConf.SESSION_LOCAL_TIMEZONE, "UTC")
```

Comment:
Why did you forcibly set it to `UTC`?

Reply:
`Instant.parse(..)` parse a UTC instant, if I don't set the timezone to UTC then it makes comparing the test input and output harder.

Reply:
If you would remove this config settings, you can replace:
```scala
  testJson(
    new Timestamp(Instant.parse("2017-01-06T10:22:03.00Z").toEpochMilli),
    TimestampType,
    JString("2017-01-06 10:22:03"))
  testJson(
    Instant.parse("2017-05-30T10:22:03.00Z"),
    TimestampType,
    JString("2017-05-30 10:22:03"))
```
by
```scala
  testJson(
    Timestamp.valueOf("2017-01-06 10:22:03.00"),
    TimestampType,
    JString("2017-01-06 10:22:03"))
  testJson(
    Timestamp.valueOf("2017-05-30 10:22:03.00").toInstant,
    TimestampType,
    JString("2017-05-30 10:22:03"))
```

Reply:
+1